lib/datasource/maven.js

```diff
@@ -0,0 +1,123 @@
+const _ = require('lodash');
+const got = require('got');
+const url = require('url');
+const fs = require('fs-extra');
+const xmlParser = require('fast-xml-parser');
+
+module.exports = {
+  getPkgReleases,
+  resetCache,
+};
+
+let map = new Map();
+
+function resetCache() {
+  map = new Map();
```

Comment:
I think this pattern came from the npmjs datasource. The reason for having it there was because the same package may be queried 10's of times for a single repo, so we added it as an in-memory cache for each run/repo. If this is unlikely to happen for maven then we could simply by leaving it out.